src/pages/SudokuPage.js

```diff
@@ -2,10 +2,12 @@ import "./SudokuPage.css";
 import SudokuGrid from "../components/SudokuGrid/SudokuGrid.js";
 import NumberInputField from "../components/NumberInputField.js";
 import ValidateBtn from "../components/ValidateBtn.js";
-import { useEffect, useState } from "react";
+import cloneMatrix from "../utility/cloneMatrix.js";
+import { useEffect, useState, useRef } from "react";
 
 export default function SudokuPage() {
-  const [numbers, setNumbers] = useState([[], [], [], [], [], [], [], [], []]);
+  const [sudoku, setSudoku] = useState(Array(9).fill([]));
+  const initialSudoku = useRef();
```

Comment:
Wow, when did you learn this? ;)

As the initial sudoku will never be rendered, it's okay to use a ref. I'm just positively surprised!

Reply:
Haha, thank you. I was reading about using variables in react and found out about useRef.
Also Martin confirmed me yesterday that a state, as a variable, is only used if something should re-render. So I changed it.